handlers/api-handler.ts

```diff
@@ -163,12 +154,33 @@ export function ApiHandler<
 				queryBody ?? {},
 			);
 
-			const validatedEvent: ValidatedApiEvent<B, Q, A, P> = {
+			const parsedEvent: ParsedApiEvent<B, Q, P> = {
 				...event,
 				input: {
 					body: validatedBody,
 					query: validatedQuery,
 					path: validatedParameters,
+				}
+			};
+
+			let auth = undefined as unknown as A;
+			if (authorizer) {
+				try {
+					const authResult = authorizer(parsedEvent, options);
+					if (authResult) {
+						auth = authResult;
+					} else {
+						return FailedResponse('Unauthorized', { statusCode: 403 });
+					}
+				} catch {
+					return FailedResponse('Unauthorized', { statusCode: 403 });
```

Comment:
This test is worth adding to ensure if the authorizer function throws an error it returns 403.